### PR TITLE
Fix unexpected exception handling

### LINKED ISSUE #540: Unexpected exception handling
**🐛Bug**

First, thanks for this awesome project!
There is an unexpected exception handling when parsing the image path.

https://github.com/fepegar/torchio/blob/bf0a86fa5832b7236932a6e0cdd0c19c8a6ce724/torchio/data/image.py#L380-L383

I think the implementation here may be better if it is in the following way. Can I send a PR for this?
Is there any other concern? :)

```python
if isinstance(path, Iterable):
    return [self._parse_single_path(p) for p in path]
return self._parse_single_path(path)
```

**To reproduce**

```python
import torchio as tio

tio.Image(0)
```

**Expected behavior**

```python-traceback
TypeError: expected str, bytes or os.PathLike object, not int
```

**Actual behavior**

```python-traceback
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/siahuat0727/.local/lib/python3.8/site-packages/torchio/data/image.py", line 141, in __init__
    self.path = self._parse_path(path)
  File "/home/siahuat0727/.local/lib/python3.8/site-packages/torchio/data/image.py", line 383, in _parse_path
    return [self._parse_single_path(p) for p in path]
TypeError: 'int' object is not iterable
```

### PR DESCRIPTION
**Pull request by siahuat0727**
Fixes #540.

**Description**
Better exception handling

**Checklist**

- [x] I have read the [`CONTRIBUTING`](https://github.com/fepegar/torchio/blob/master/CONTRIBUTING.rst) docs and have a developer setup (especially important are `pre-commit`and `pytest`)
- [x] Non-breaking change (would not break existing functionality)
- [ ] Breaking change (would cause existing functionality to change)
- [ ] Tests added or modified to cover the changes
- [x] Integration tests passed locally by running `pytest`
- [ ] In-line docstrings updated
- [ ] Documentation updated, tested running `make html` inside the `docs/` folder
- [x] This pull request is ready to be reviewed
- [x] If the PR is ready and there are multiple commits, I have [squashed them and force-pushed](https://www.w3docs.com/snippets/git/how-to-combine-multiple-commits-into-one-with-3-steps.html#force-pushing-commits-7)

### CODE REVIEW

File: torchio/data/image.py [hunk 1]
```diff
@@ -377,10 +378,10 @@ def _parse_path(
             ) -> Optional[Union[Path, List[Path]]]:
         if path is None:
             return None
-        if isinstance(path, (str, Path)):
-            return self._parse_single_path(path)
-        else:
+        if isinstance(path, list):
```
Comment (fepegar) [inline torchio/data/image.py:381]: Did you mean `Iterable` here?

**Review by fepegar — CHANGES_REQUESTED** [review]

File: torchio/data/image.py (hunk 1 above)
Comment (siahuat0727) [reply torchio/data/image.py:381]: Ops... I changed it at first to try something else and then forgot about it.
Can I use `git commit --amend` and then force push for this situation?

Comment (fepegar) [reply torchio/data/image.py:381]: You can just fix this in a new commit, don't worry about squashing or force-pushing!

**Comment by fepegar** [comment]
Thanks again, @siahuat0727.

@all-contributors please add @siahuat0727 for bug

**Pull request merged**